aten/src/ATen/native/Convolution.cpp

```diff
@@ -146,7 +148,31 @@ auto ConvParams::use_cpu_depthwise3x3_winograd(
 #endif
 }
 
-auto ConvParams::use_cudnn(const at::Tensor& input) const -> bool {
+auto ConvParams::needs_64bit_indexing_no_split(const at::Tensor& input, const at::Tensor& weight) const -> bool {
+  constexpr int64_t int_max = std::numeric_limits<int>::max();
+  int64_t numel_input = input.numel();
+  // empty input
+  if (numel_input == 0) {
+    return false;
+  }
+  // input size can not be reduced to the range of int by splitting the batch dim
+  int64_t n = input.size(0);
+  if (numel_input / n > int_max) {
+    return true;
+  }
+  // output size can not be reduced to the range of int by splitting the batch dim
+  int64_t outsize = weight.size(0);
+  for (size_t d = 2; d < input.dim(); ++d) {
+    auto kernel = dilation[d - 2] * (weight.size(d) - 1) + 1;
```

Comment:
If params.transposed is true for this convolution, the formula for output size will be different.